src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
+  //when it's attached
+  var observer = null;
+  if (!document.contains(el)) {
```

Comment:
as an alternative, you can do:

```javascript
var contains = document.contains || document.body.contains;
if (!contains(el)) {
...
}
```

PhantomJS framework can sometimes surface issues like this; but, the pros still outweigh its cons, so we are still sticking to it. Also, any such issue will affect people trying to render views in headless mode, so it would be great if we address this even though it is a weird workaround.